public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts

```diff
@@ -299,16 +314,28 @@ export default class AzureLogAnalyticsDatasource extends DataSourceWithBackend<
     return quotedValues.join(',');
   }
 
-  getDefaultOrFirstWorkspace() {
+  async getDefaultOrFirstSubscription(): Promise<string | undefined> {
+    if (this.defaultSubscriptionId) {
+      return this.defaultSubscriptionId;
+    }
+    return this.getSubscriptions().then((subs) => first(subs.map((s) => s.value)));
```

Comment:
`.then((subs) => subs[0]?.value)`?

Reply:
Don't know,  IMHO `first`/`map` is much more readable than `0]?.`

Reply:
Changed.